### PR TITLE
[RQ-1150] fix: rules table analytics and user attributes

### PR DESCRIPTION
**Pull request by rohanmathur91**
Closes issue: 

## ✅ Checklist:

- [ ] Make sure linting and unit tests pass.
- [ ] No install/build warnings introduced.
- [ ] Verified UI in browser.
- [ ] For UI changes, added/updated analytics events (if applicable).
- [ ] For changes in extension's code, manually tested in Chrome and Firefox.
- [ ] For changes in extension's code, both MV2 and MV3 are covered.
- [ ] Added/updated unit tests for this change.
- [ ] Raised pull request to update corresponding documentation (if already exists).

### CODE REVIEW

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx [hunk 1]
```diff
@@ -115,17 +117,27 @@ const useRuleTableColumns = (options: Record<string, boolean>) => {
...
+            source="rule_list_status_switch"
+          >
+            <Switch
+              size="small"
+              checked={isRuleActive}
+              data-tour-id={index === 0 ? "rule-table-switch-status" : null}
+              onChange={(checked: boolean, e) => {
+                e.stopPropagation();
+                handleStatusToggle([rule]);
+              }}
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx:138]: Remove this onChange, `PremiumFeature` wrapper will handle it if toggle is allowed.

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx [hunk 2]
```diff
@@ -37,47 +39,55 @@ const useRuleTableActions = () => {
...
-      const status = checked ? RuleObjStatus.ACTIVE : RuleObjStatus.INACTIVE;
+      const status = rule.status === RuleObjStatus.ACTIVE ? RuleObjStatus.INACTIVE : RuleObjStatus.ACTIVE;
       changeRuleStatus(status, rule);
     });
   };
 
   const changeRuleStatus = (newStatus: RuleObjStatus, rule: RuleObj) => {
     // TODO: Handle Group status toggle
-    console.log({ rule, user });
+    console.log("group", { rule, user });
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx:54]: remove log

File: app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx [hunk 3]
```diff
@@ -37,47 +39,55 @@ const useRuleTableActions = () => {
     setSelectedRows([]);
   }, [setSelectedRows]);
 
-  const handleStatusToggle = (rules: RuleTableDataType[], checked: boolean) => {
-    console.log("handleStatusToggle", { rules, checked });
+  const handleStatusToggle = (rules: RuleTableDataType[]) => {
+    console.log("handleStatusToggle", { rules });
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesTable/hooks/useRuleTableActions.tsx:43]: remove log

File: app/src/features/rules/components/RulesList/components/RulesListIndex/hooks/useFetchAndUpdateRules.tsx [hunk 4]
```diff
@@ -29,44 +31,38 @@ const useFetchAndUpdateRules = ({ setIsLoading }: Props) => {
...
-      //   TRACKING.ATTR.NUM_ACTIVE_RULES,
-      //   rules.filter((rule) => rule.status === GLOBAL_CONSTANTS.RULE_STATUS.ACTIVE).length
-      // );
-      // submitAttrUtil(TRACKING.ATTR.NUM_GROUPS, groups.length);
-      // submitAttrUtil(
-      //   TRACKING.ATTR.NUM_ACTIVE_GROUPS,
-      //   groups.filter((group) => group.status === GLOBAL_CONSTANTS.GROUP_STATUS.ACTIVE).length
-      // );
+      const ruleTypes = rules.reduce((result, { ruleType }) => result.add(ruleType), new Set());
+      const numRuleTypes = parseInt(window.localStorage.getItem("num_rule_types") || "0");
```
Comment (RuntimeTerror10) [inline app/src/features/rules/components/RulesList/components/RulesListIndex/hooks/useFetchAndUpdateRules.tsx:52]:
```suggestion
      const numRuleTypes = parseInt(window.localStorage.getItem("num_rule_types") || 0);
```

Comment (rohanmathur91) [reply app/src/features/rules/components/RulesList/components/RulesListIndex/hooks/useFetchAndUpdateRules.tsx:52]: `parseInt` expects a string.

**Review by RuntimeTerror10 — APPROVED** [review]

**Pull request merged**